### PR TITLE
Issue573

### PR DESCRIPTION
**Pull request by russell-d-e**
#573 and #576 

Added ontology for Test Procedure and test Logs.

Also created new Ontology for Hardware Components

### CODE REVIEW

**Comment by AbhaMoitra** [comment]
@russell-d-e : We have the following
  " TEST_PROCEDURE is a COLLECTION of TEST_STEP which in turn is a COLLECTION of TEST"
and associated with these we have respectively
  " TEST_LOG, TEST_RECORD and TEST_RESULT"
While the "note" tells us that TEST_LOG is for TEST_PROCEDURE" etc. that may not be something people have in their mental model and they may not have easy access to the "note". 
Q. So, do you think we should rename TEST_LOG and TEST_RECORD?
Q. We have 3 levels in this hierarchy - just checking. You know this much better.

We have the following
	TEST_RECORD (note "a TEST_RECORD is a collection of TEST_RESULTs assoicated with a TEST_STEP") is a type of COLLECTION.
		content of TEST_RECORD only has values of type TEST_RESULT.
		nextRecord (note "The next TEST_RECORD that is the next in a sequence " ) describes TEST_STEP with a single value of type TEST_RECORD.
		logs (note "The next TEST_STEP from which the TEST_RECORD is captured" ) describes TEST_STEP with a single value of type TEST_STEP.
		logs is a type of wasImpactedBy.

- correct the spelling of "assoicated"
- did we want domain of nextRecord to be TEST_RECORD? Do we typically navigate by going to "next test step" and looking at the results (in which case the property name could be "record" instead of "nextRecord") or we go from "result" to "next result"?
- also we have log: TEST_STEP -> TEST_STEP?

**Comment by kityansiu** [comment]
Putting a comment here in case git merge doesn't resolve it. Be sure to remove from TESTING.sadl the following lines:

under TEST
`producedBy (note "TEST_DEVELOPMENT activity(s) this test belongs to") describes TEST with values of type ACTIVITY.`
`producedBy is a type of wasGeneratedBy.`

under TEST_RESULT
`executedBy (note "ACTIVITY(s) that created this test result") describes TEST_RESULT with values of type ACTIVITY.`
`executedBy is a type of wasGeneratedBy.`

**Comment by russell-d-e** [comment]
@kityansiu and @AbhaMoitra 

I think I have merged all the changes back from master into this branch. could you do one more quick check just because the merge got a bit messy. If it looks good we can merge and remove the branch.

File: RACK-Ontology/OwlModels/.gitignore
Comment (tuxji) [inline RACK-Ontology/OwlModels/.gitignore:1]: Please remove this file.  We don't need it since 1) ignoring OWL files already in git doesn't work, and 2) when the time comes, we'll remove OWL files from git and put *.owl in RACK/.gitignore at the root level.

**Review by tuxji — COMMENTED** [review]
Looks good to me, although I'll defer to Kit and Abha for approval.  I just wanted to say that I hope we'll stop updating OWL and CDR files soon and remove them from git since the CI jobs automatically generate new OWL and CDR files now (meaning that the files in git don't make it to the rack-box dev image anyway).

**Comment by tuxji** [comment]
@russell-d-e, another suggestion too.  As you said, this branch has a messy git history because of its multiple merges and commits.  When I look at the "Files Changed" tab, I see a pretty small changeset (changes to a handful of SADL files) that could easily fit into a single commit with a single commit message.  GitHub has a dropdown button which allows you to select how to merge the pull request into the main branch: 1) Create a merge commit; 2) Squash and merge; 3) Rebase and merge.  You want to pick 2) Squash and merge, which will squash the branch's messy git history into a single commit.  I've used GitHub's Squash and merge button before and the UI works well.  After you press the button, the multiple commits become one commit and you're shown a text editor pane which allows you to edit the combined commit messages however you want.  You can clean up that commit message so you end up with a single commit with a single commit message that mentions only the changes to the SADL files.

**Review by kityansiu — APPROVED** [review]

File: RACK-Ontology/ontology/HARDWARE.sadl
Comment (kquick) [inline RACK-Ontology/ontology/HARDWARE.sadl:23]: "PARTITION",  "HWCOMPONENTS"

Comment (glguy) [reply RACK-Ontology/ontology/HARDWARE.sadl:23]: Fixed in #620

**Pull request merged**
